src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsEventNotificationConfig.java

```diff
@@ -91,16 +92,13 @@ public static Builder builder() {
     public ValidationResult validate() {
         ValidationResult validation = new ValidationResult();
 
-        URI webhookUri;
         try {
-            webhookUri = new URI(webhookUrl());
-            if (webhookUri.getHost().toLowerCase().contains("office")) {
-                if (!TEAMS_PATTERN.matcher(webhookUrl()).find()) {
-                    validation.addError(FIELD_WEBHOOK_URL, INVALID_TEAMS_URL_ERROR_MESSAGE);
-                }
+            final URI webhookUri = new URI(webhookUrl());
+            if (!webhookUri.getHost().toLowerCase().contains("office") ||
+                    !TEAMS_PATTERN.matcher(webhookUrl()).find()) {
+                validation.addError(FIELD_WEBHOOK_URL, INVALID_TEAMS_URL_ERROR_MESSAGE);
             }
-
-        } catch (Exception ex) {
+        } catch (URISyntaxException ex) {
```

Comment:
This now handles an incorrect webhook perfectly, but I noticed here that if you use something crazy that's not a valid URI, you get a `NullPointerException` at line 97 which isn't caught anymore. The one downside is that you end up with an 'Unknown error' at the UI instead of explicitly getting a pretty 'Invalid URL' message with field highlighting.

Same with the slack notification.

Reply:
Yep, fixed that and reverted back to a generic Exception

Reply:
> Yep, fixed that and reverted back to a generic Exception

You are a beast, I can't even think as fast as you crank out commits